### ISSUE TITLE
Problem with transaction and wallet not showing the right amounts!!

### ISSUE THREAD

**Issue by bowsfixer** [decred/decrediton#449]
I've recreated my initial MacOs wallet in the web wallet from seed and it says 0!!!! The transaction that appear in the web wallet only partially correspond  to the ones I made yesterday.  I'm kinda freaking out here.  Anyone out there that can help?

**Comment by jcvernaleo** [comment]
@bowsfixer not entirely sure I understand the situation you are describing.  You made a wallet with decrediton on OSX and made some transactions there.  Does decrediton show you what you think is the correct balance?

Then you made a webwallet using the same seed as your decrediton wallet.  That one does not show the balance correctly (and doesn't show all your transactions).

Is that a correct summary of your situation?  If so, that sounds like an issue with the webwallet, not decrediton and should be reported at https://github.com/decred/copay/issues.  copay does have know issues in calculating balances though and we have a big update of it in the works (see this PR for details on the update: https://github.com/decred/copay/pull/63).

If I'm misunderstanding the issue, could you please try to clarify?

**Comment by bowsfixer** [comment]
Thank you for your quick reply. 

Yes I initially used decrediton on MacOs for a total of 2 transactions.  First one was receive and the second one to send. 

Yes my balance shows correctly the remaining DCR but the receiver still is waiting for his payment... 

The only reason I tried the web wallet was because decrediton was giving me error messages at start up and I believe at shut down so I wanted to double check that everything went well with this transaction. Once imported, the transaction was very weird (2 receivers) and the shows balance was 0 now.  But as you mentioned that's probably another issue with the web wallet itself.

If I disregard the 0 DCR balance in the web wallet for now until the update, I still have not been able to send the amount to the receiver.

**Comment by jcvernaleo** [comment]
Were you able to look up the transaction on the block explorer? https://mainnet.decred.org/

**Comment by jcvernaleo** [comment]
Also, what error message did decrediton give you.  And was decrediton fully synced before you started doing this?

**Comment by bowsfixer** [comment]
-tried block explorer... not sure what I should be looking for
-Yes decrediton fully synced. 
-Error at startup says: "Connection to RPC server failed: dial tcp [::1]:9119: getsockopt: connection refused. You may need to edit /Users/xxxxxx13/Library/Application Support/decrediton/config.json and try again"

**Comment by bowsfixer** [comment]
- Actually looking again with block explorer, the amount is not right.
- Actually to remember well decrediton was NOT synched before I received the initial amount but then when I tried sending, I believe it was.  It was 6 hours later. How can I be sure?

**Comment by bowsfixer** [comment]
Should send the block number so you can see for yourself? I'm kinda freaking out now, I am not an expert but it looks like my transaction was sent to another address first and the wrong amount and then split in 2 adresses (only the first address is correct and amount is correct too).

**Comment by jcvernaleo** [comment]
If you use slack, it might be helpful for you to join the #support channel on https://slack.decred.org/ since that allows for quicker back and forth (and there are more people who can help).

But anyway...

The 2 address split is normal.  When you make a transaction, it usally has to send more and then send the rest back to you as change (bitcoin works the same way).  I can show you an example on testnet (using the command line since it is easier to copy and paste).  I sent 10 dcr from one wallet to another wallet:
```
jcv@decred1:~$ dcrctl --testnet --wallet sendtoaddress TsT65tXVSuxt5KSCcFpAVSZWu93mDD1BA5B 10
73a6b5372f5b3492c3621cdc116170f558f4b1f53579af6f29dcadd4e8bd907e
```
Now I look up that transaction `73a6....` on the blockexplorer:
https://testnet.decred.org/tx/73a6b5372f5b3492c3621cdc116170f558f4b1f53579af6f29dcadd4e8bd907e
As you can see there, there is the correct amount (10) going to the address in the command above and 44dcr going to `TsiRLEJZLmnTG1hFv9dhQs8EcKWvF6o6RSw`
If I check out that address:
```
jcv@decred1:~$ dcrctl --testnet --wallet validateaddress TsiRLEJZLmnTG1hFv9dhQs8EcKWvF6o6RSw | grep ismine
  "ismine": true,
```
I can see that it is my address.  decrediton does not have that validateaddress function but I'm hoping to add it in to the next release.

**Comment by bowsfixer** [comment]
Thank you so much, that reassures me. I'm going to wait for an answer of the receiver. 
Why Changelly site is delaying the reception .  Usually they are great.  Sorry if I've been alarming, all this may just be delays from their side.

**Comment by bowsfixer** [comment]
@jcvernaleo I have an update regarding my issue ( which is btw the same regarding case #448 ). 
Changelly.com specifies when sending them coins for exchange that it needs to be done in ONE transaction. I just sent them the 90 DCR using decrediton not knowing that my transaction was going to be sent in 2 transactions (i.e. sending 97.74379614 DCR and then getting back my change 7.74379614 DCR). Their system did not recognize my payment... they have my funds but refuse 'till now to exchange them.   
https://mainnet.decred.org/tx/0ce0b4434625aea267d63360e5bd7719858d64fb426544142c0d48b6150e9621

What do you suggest doing now? and how could have I prevented this situation? Thank you

**Comment by alexlyp** [comment]
That is totally on Changelly's end.  There is only 1 transaction. (the one you linked).  I dont think we plan on adding a special case to construct perfectly sized utxos.

**Comment by chappjc** [comment]
Geez, to get a transaction with a perfectly sized output would require an additional transaction.  Surely that's not what Changelly is requiring here.  You did do it in one transactoin, so you've done nothing wrong or even unusual.


Expanding the transaction on block explorer (with the + icon) shows the output to an address starting with "Dsf78mfF" so if that was the address Changelly gave you, then they do have your funds.  Moreover, they've since moved those funds.

**Comment by bowsfixer** [comment]
Yes that was their address, so I'm happy It wasn't a mistake on my side or even from decrediton. Not happy at all, that they _held_ that transfer for 2 days (far from the 30 minutes normal transaction time) misleading "waiting for payment" as if  they haven't received the transfer. Finally Changelly sent the equivalent in LTC today (actually about 4 LTC less then the initial transaction rate). I have my doubts they would have added the extra 4 LTC if the rates would had turned out the other way to my advantage... 

It's so hard to get a response from them, (unlike here) it becomes very frustrating. Sending those coins without have some simple confirmation or acknowledgement of reception is stressful. Don't get me wrong I think Changelly is a great site but the support group is nil.

Thank you for the help and comments.  I may try to ask for a refund for difference of rates caused by the delay. Hope writing and expanding this ticket may help others eventually sort out similar problems with them.

**Comment by jcvernaleo** [comment]
@bowsfixer I appreciate you giving so many details here.  Having all the details written down can only help in the long run.

**Comment by chappjc** [comment]
I'm sure they will sort out the issue.  Their wallet actually spent the funds you sent them _less than 6 hours_ after receiving them, so delaying the transfer for 2 days is not remotely justifiable.  I suspect their transaction notification system had (has?) a glitch.  That said, I've tested Changelly for going back and forth between DCR and BTC shortly after Decred was added, and it worked for me.  Hopefully it will be smoother for all users in the future.
